### ISSUE TITLE
SQL Syntax Error when Update to 3.1.0

### ISSUE THREAD

**Issue by alexdhika** [mautic/mautic#9180]
> php bin/console doctrine:migration:migrate

++ adding generated column generated_sent_date
-> ALTER TABLE ema_email_stats ADD generated_sent_date DATE AS (CONCAT(YEAR(date_sent), "-", LPAD(MONTH(date_sent), 2, "0"), "-", LPAD(DAY(date_sent), 2, "0"))) COMMENT '(DC2Type:generated)';
            ALTER TABLE ema_email_stats ADD INDEX `ema_generated_sent_date_email_id`(generated_sent_date, email_id)

In AbstractMySQLDriver.php line 79:
                                                                                                      
  An exception occurred while executing 'ALTER TABLE ema_email_stats ADD generated_sent_date DATE AS  
   (CONCAT(YEAR(date_sent), "-", LPAD(MONTH(date_sent), 2, "0"), "-", LPAD(DAY(date_sent), 2, "0")))  
   COMMENT '(DC2Type:generated)';                                                                     
              ALTER TABLE ema_email_stats ADD INDEX `ema_generated_sent_date_email_id`(generated_sen  
  t_date, email_id)':                                                                                 
                                                                                                      
  You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server ver  
  sion for the right syntax to use near 'ALTER TABLE ema_email_stats ADD INDEX `ema_generated_sent_d  
  ate_email_id`(generat' at line 2                                                                    
                                                                                                      

In MysqliStatement.php line 86:
                                                                                                      
  You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server ver  
  sion for the right syntax to use near 'ALTER TABLE ema_email_stats ADD INDEX `ema_generated_sent_d  
  ate_email_id`(generat' at line 2

**Comment by andrelug** [comment]
I have the same exact error here as well. I'd like to ask if someone know the solution, please. I thank you in advance

**Comment by alexdhika** [comment]
Sill error in mautic 3.1.1

**Comment by RCheesley** [comment]
Hi folks, I'm guessing that this is related to https://github.com/mautic/mautic/issues/9129#issuecomment-707708797 so I'll close this issue down in favour of the discussions there.

**Comment by mautibot** [comment]
This issue has been mentioned on **Mautic Community Forums**. There might be relevant details there:

https://forum.mautic.org/t/error-updating-mautic-2-16-4-to-3-2/17368/1

**Comment by ttencate** [comment]
@RCheesley 
> Hi folks, I'm guessing that this is related to [#9129 (comment)](https://github.com/mautic/mautic/issues/9129#issuecomment-707708797) so I'll close this issue down in favour of the discussions there.

I don't think that is the same issue. The error message quoted in #9129 is quite different: that one is throwing an error on foreign key constraints, but this one is a bona fide SQL syntax error.

`bin/console doctrine:migrations:migrate` reports the same error.

I copied and pasted the SQL into a mysql shell, where it works fine, although I had to add a semicolon at the end:

```sql
ALTER TABLE email_stats ADD generated_sent_date DATE AS (CONCAT(YEAR(date_sent), "-", LPAD(MONTH(date_sent), 2, "0
"), "-", LPAD(DAY(date_sent), 2, "0"))) COMMENT '(DC2Type:generated)';
            ALTER TABLE email_stats ADD INDEX `generated_sent_date_email_id`(generated_sent_date, email_id)
```

However, note that there are two statements in one. Maybe the shell is fine with that, but the API used by the Doctrine migrations chokes on it? It could be a Doctrine bug but I couldn't find anything relevant in [their issue tracker](https://github.com/doctrine/orm/issues).

Anyway, after running these commands in the mysql shell, I was able to run `bin/console doctrine:migrations:migrate` without any further errors.
